forest/daemon/src/main.rs

```diff
@@ -107,7 +107,7 @@ fn main() {
     // Run forest as a daemon if no other subcommands are used. Otherwise, run the subcommand.
     match opts.to_config() {
         Ok((cfg, path)) => {
-            let (loki_task, ..) = logger::setup_logger(&cfg.log, &opts);
+            let (loki_task, _file_logger_worker_guard) = logger::setup_logger(&cfg.log, &opts);
```

Comment:
`_file_logger_worker_guard` will be dropped when we detach. Is that okay?

Reply:
What are the benefits and drawbacks of the blocking and non-blocking loggers?

Reply:
The non blocking one queues log entries in memory and uses a dedicated thread to write files, while a blocking one is just like the console logger, writes to the file (buffer) directly

Reply:
How we always use the blocking logger, then? I'm concerned that having different logic for the detach and non-detach modes will lead to unexpected behavior.

Reply:
Sounds good, I don't really think it will introduce too big overhead

Reply:
Fix pushed